src/Adapter/Address/CommandHandler/EditCustomerAddressHandler.php

```diff
@@ -57,12 +60,32 @@ public function handle(EditCustomerAddressCommand $command): void
                 throw new AddressConstraintException('Address contains invalid field values');
             }
 
-            if (false === $address->update()) {
+            // The address is used by an order so it is not edited directly, instead a copy is created and
+            if ($address->isUsed()) {
+                // Get a copy of current address
+                $old_address = new Address($address->id);
+
+                // Reset ID to force recreating a new address
+                $address->id = $address->id_address = null;
+
+                // We consider this address as necessarily NOT deleted, in case you were editing a deleted address
+                // from an order then the newly edited address should not be deleted, so that you can select it
+                $address->deleted = 0;
+                if (false === $address->save()) {
+                    throw new CannotAddAddressException(sprintf('Failed to add new address "%s"', $command->getAddress()));
+                }
+                // Soft delete the former address
+                if (false === $old_address->delete()) {
```

Comment:
delete $address to be consistent and save the copy instead

Reply:
As explained above `$address` contains the edited data so it makes more sense to save/insert this one Besides `$address->delete()` sets deleted to true and then performs an update. So it would erase the previous data and save the edited data in the deleted row.

But I understand that the code is not really instinctive.. Maybe if we renamed `$address` into `$editedAddress` it would be easier to understand?